### PR TITLE
[runtime] Add better control over the random hashing seed

### PR DESCRIPTION
**Pull request by lorentey**
This changes the runtime to look for the `SWIFT_DETERMINISTIC_HASHING` environment variable, and, if it is set to `1` during process startup, arrange to disable randomized hashing in the stdlib.

This mechanism replaces the old mutable `_Hasher._secretKey` property with a read-only `_Hasher._seed` var, and adds a new `_Hasher._isDeterministic` boolean property indicating whether `Set` and `Dictionary` should ensure repeatable ordering.

The seed generator is updated to make better use of the system-provided RNGs.

Swift's lit config is updated to run executable tests with deterministic hashing enabled by default, on all platforms.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci please test

File: stdlib/public/core/Hashing.swift [hunk 1]
```diff
@@ -27,30 +27,14 @@ import SwiftShims
...
   // FIXME(hasher): Remove
   @_inlineable // FIXME(sil-serialize-all)
   @_versioned
   @_transparent
   internal static func getExecutionSeed() -> UInt64 {
     // FIXME: This needs to be a per-execution seed. This is just a placeholder
     // implementation.
-    let seed: UInt64 = 0xff51afd7ed558ccd
-    return _HashingDetail.fixedSeedOverride == 0 ? seed : fixedSeedOverride
+    return 0xff51afd7ed558ccd
```
Comment (milseman) [inline stdlib/public/core/Hashing.swift]:
```
return 0xff51afd7ed558ccd // Chosen by fair dice role. Guaranteed to be random
```

**Review by milseman — APPROVED** [review]
LGTM

File: stdlib/public/core/Hashing.swift (hunk 1 above)
Comment (moiseev) [reply stdlib/public/core/Hashing.swift]: 0xffffffffffffffff sided dice.

**Pull request merged**